jina/serve/runtimes/worker/__init__.py

```diff
@@ -100,8 +100,8 @@ async def async_cancel(self):
         self.logger.debug('Cancel WorkerRuntime')
 
         # 0.5 gives the runtime some time to complete outstanding responses
-        # this should be handled better, 0.5 is a rather random number
-        await self._grpc_server.stop(0.5)
+        # this should be handled better, 1.0 is a rather random number
+        await self._grpc_server.stop(1.0)
```

Comment:
there is no way to handle this better? even without timeout?

Reply:
I dont think there is. What this does is "Stop accepting new requests" -> "wait for grace period" -> "cancel non completed requests".
We have no control over this otherwise and I think the general mechanism is fine. Just the grace period is a random value. Its not needed for this PR, we can also leave it as it was. I was just thinking 1s is a bit better than 0.5s.